common/protos/identity/agent.proto

```diff
@@ -14,17 +15,38 @@ service IINAgent {
   // Requesting network unit's state from a foreign IIN agent.
   rpc RequestIdentityConfiguration(NetworkUnitIdentity) returns (common.ack.Ack){}
   // Handling network unit's state sent by a foreign IIN agent.
-  rpc SendIdentityConfiguration(NetworkUnitIdentity) returns (common.ack.Ack){}
-  // user or agent triggers a flow to collect signatures attesting an
-  // external/foreign network unit's state and recording to ledger
-  rpc FlowAndRecordAttestations(NetworkUnitIdentity) returns (common.ack.Ack) {}
+  rpc SendIdentityConfiguration(AttestedMembership) returns (common.ack.Ack){}
   // Requesting attestation from a local IIN agent.
-  rpc RequestAttestation(NetworkUnitIdentity) returns (common.ack.Ack){}
+  rpc RequestAttestation(AttestedSecurityDomain) returns (common.ack.Ack){}
   // Handling attestation sent by a local IIN agent.
-  rpc SendAttestation(NetworkUnitIdentity) returns (common.ack.Ack){}
+  rpc SendAttestation(AttestedSecurityDomain) returns (common.ack.Ack){}
 }
 
+// Unique identifier for a unit of a network that runs an IIN agent
 message NetworkUnitIdentity {
   string network_id = 1;
```

Comment:
Do you want to include a timestamp here (or somewhere else) or some id? so during a flow, that `id` can be used to track which particular request it is.
During async processing, the iin-agent will be stateless (i.e. not remember for which particular request it is getting response), and thus the information that is exchanged need to have some state/id information.
I thought of this structure because, this message is what is sent during the first step of the whole flow. So during first step only we can assign some timestamp or id.

Reply:
Yes, I can add a `nonce` field. Or do you prefer `timestamp`?